### PR TITLE
Refactor/activity class

### PR DESCRIPTION
**Pull request by PaigeVannelli**
## Pull Request

<br>

### Description
Change fetch structure so that all of the fetching is done in one function and the data is returned in one object 
Connect fetch request to scripts
Set up handler functions to display activity data, hydration data and sleep data 

* Is this a feature or a fix?
Fix - refactor how we are accessing our data 

* Where should the reviewer start?
Review APICalls.js and scripts.js to review startApp function 

* What functionalities do these changes effect? Why were these changes made? What was the motivation behind these changes?
Updates our display data 

* What outside features or research did you use to implement this fix?

* How should this be tested?

<br>

***

#### Please review considerations below:

- [ ] Follows Turing Style Guidelines

- [ ] Changes generate no new warnings or errors

- [ ] Checked code for and corrected any spelling errors

- [ ] README has been updated(if applicable)

### CODE REVIEW

File: src/APICalls.js [hunk 1]
```diff
@@ -1,40 +1,43 @@
...
+            return userData
+        })
+        .catch(err => console.log(err.message))
+    
+    let hydrationData = fetch("http://localhost:3001/api/v1/hydration")
+        .then(response => response.json())
+        .then(hydrationData => {
+            return hydrationData
+        })
+        .catch(err => console.log(err.message))
```
Comment (cbmackintosh) [inline src/APICalls.js:16]: Make a note to add real error message for users - improve error handling on these functions...

File: src/scripts.js [hunk 2]
```diff
@@ -120,7 +159,8 @@ function makeRandomDate(userStorage, id, dataSet) {
 
 function addHydrationInfo(id, hydrationInfo, dateString, userStorage, laterDateString) {
   hydrationToday.insertAdjacentHTML('afterBegin', `<p>You drank</p><p><span class="number">${hydrationInfo.calculateDailyOunces(id, dateString)}</span></p><p>oz water today.</p>`);
-  hydrationAverage.insertAdjacentHTML('afterBegin', `<p>Your average water intake is</p><p><span class="number">${hydrationInfo.calculateAverageOunces(id)}</span></p> <p>oz per day.</p>`)
```
Comment (percworld) [inline src/scripts.js:123]: Let's refactor how we are manipulating the DOM

File: src/APICalls.js [hunk 3]
```diff
@@ -1,40 +1,43 @@
...
+    
+    let sleepData = fetch("http://localhost:3001/api/v1/sleep")
+        .then(response => response.json())
+        .then(sleepData => {
+            sleepData
+        })
+        .catch(err => console.log(err.message))
+    
+    return Promise.all([userData, hydrationData, activityData, sleepData])
+    .then(data => {
```
Comment (cbmackintosh) [inline src/APICalls.js:33]: Nice work figuring this out. Very streamlined method for fetching all necessary Data in one go.

File: src/Hydration.js [hunk 4]
```diff
@@ -1,12 +1,14 @@
...
-  constructor(hydrationData) {
+  constructor(hydrationData, userNow) {
     this.hydrationData = hydrationData;
+    this.user = userNow;
   }
-  calculateAverageOunces(id) {
-    let perDayUserHydration = this.hydrationData.filter((data) => id === data.userID);
-    return perDayUserHydration.reduce((sumSoFar, data) => {
+  calculateAverageOunces() {
+    let perDayUserHydration = this.hydrationData.hydrationData.filter((data) => this.user.id === data.userID);
```
Comment (percworld) [inline src/Hydration.js:7]: Can we assign hydrationData to a new object earlier to relieve later confusion?

**Review by percworld — COMMENTED** [review]
Inheritance: May consider making average calculating class

File: src/Activity.js [hunk 5]
```diff
@@ -1,27 +1,43 @@
...
+    // let userActivityByDate = this.activityData.find(data => this.user.id === data.userID && this.date === data.date);
     return userActivityByDate.minutesActive;
   }
-  calculateActiveAverageForWeek(id, date, userRepo) {
-    return parseFloat((userRepo.getWeekFromDate(date, id, this.activityData).reduce((acc, elem) => {
+
+  calculateActiveAverageForWeek() {
+    const totalActiveMinutes = this.userRepo.getWeekFromDate(this.date, this.user.id, this.activityData)
+    return parseFloat((totalActiveMinutes.reduce((acc, elem) => {
       return acc += elem.minutesActive;
```
Comment (cbmackintosh) [inline src/Activity.js:29]: We can consider adding a parent class to inherit methods such as calculate average. Will DRY up code a bit.

**Review by cbmackintosh — APPROVED** [review]

**Pull request merged**